### PR TITLE
make sure structs are big enough to hold all slots even if :offset is used

### PR DESCRIPTION
**Pull request by 3b**

### CODE REVIEW

File: tests/struct.lisp [hunk 1]
```diff
@@ -713,3 +713,29 @@
...
+  (f :uint8 :offset 6)
+  (g :uint32 :offset 0))
+
+(deftest struct.offset.1
+    (foreign-type-size '(:struct struct.offsets))
+  20)
+
+;; Add some tests for :offset in general
+(deftest struct.offset.2
+    (values (cffi:foreign-slot-offset '(:struct struct.offsets) 'a)
```
Comment (luismbo) [inline tests/struct.lisp:734]: No need for the package prefix, I guess. We don't use it elsewhere.

**Review by luismbo — CHANGES_REQUESTED** [review]
Nice edge case.

File: tests/struct.lisp [hunk 2]
```diff
@@ -713,3 +713,29 @@
...
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'b)
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'c)
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'd)
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'e)
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'f)
+            (cffi:foreign-slot-offset '(:struct struct.offsets) 'g))
```
Comment (luismbo) [inline tests/struct.lisp:740]: Maybe use mapcar across each slot name? 🤷‍♂️

File: tests/struct.lisp (hunk 1 above)
Comment (3b) [reply tests/struct.lisp]: oops, added it to get autodoc since i hadn't loaded the package, and forgot to remove it

File: tests/struct.lisp (hunk 2 above)
Comment (3b) [reply tests/struct.lisp]: done

**Comment by 3b** [comment]
added a few more slots to test some other properties of `:offset` as well

**Review by luismbo — APPROVED** [review]
